Mathlib/AlgebraicTopology/SimplicialSet/StdSimplex.lean

```diff
@@ -266,6 +259,59 @@ lemma range_δ {n : ℕ} (i : Fin (n + 2)) :
   rw [Subcomplex.range_eq_ofSimplex]
   exact ofSimplex_yonedaEquiv_δ i
 
+/-- The standard simplex identifies to the nerve to the preordered type
+`ULift (Fin (n + 1))`. -/
+def isoNerve (n : ℕ) :
+    (Δ[n] : SSet.{u}) ≅ nerve (ULift.{u} (Fin (n + 1))) :=
+  NatIso.ofComponents (fun d ↦ Equiv.toIso (objEquiv.trans
+    { toFun f := (ULift.orderIso.symm.monotone.comp f.toOrderHom.monotone).functor
+      invFun f :=
+        SimplexCategory.Hom.mk
+          (ULift.orderIso.toOrderEmbedding.toOrderHom.comp f.toOrderHom)
+      left_inv _ := by aesop
+      right_inv _ := by rfl }))
+
+@[simp]
+lemma isoNerve_hom_app_apply {n d : ℕ}
+    (s : (Δ[n] _⦋d⦌)) (i : Fin (d + 1)) :
+    ((isoNerve.{u} n).hom.app _ s).obj i = ULift.up (s i) := rfl
+
+@[simp]
+lemma isoNerve_inv_app_apply {n d : ℕ}
+    (F : (nerve (ULift.{u} (Fin (n + 1)))) _⦋d⦌) (i : Fin (d + 1)) :
+    (isoNerve.{u} n).inv.app _ F i = (F.obj i).down := rfl
+
+lemma mem_nonDegenerate_iff_strictMono {n d : ℕ} (s : (Δ[n] : SSet.{u}) _⦋d⦌) :
+    s ∈ Δ[n].nonDegenerate d ↔ StrictMono s := by
+  rw [← nonDegenerate_iff_of_mono (isoNerve n).hom,
+    PartialOrder.mem_nerve_nonDegenerate_iff_strictMono]
+  rfl
+
+lemma mem_nonDegenerate_iff_mono {n d : ℕ} (s : (Δ[n] : SSet.{u}) _⦋d⦌) :
+    s ∈ Δ[n].nonDegenerate d ↔ Mono (objEquiv s) := by
+  rw [mem_nonDegenerate_iff_strictMono,
+    SimplexCategory.mono_iff_injective]
+  refine ⟨fun h ↦ h.injective, fun h ↦ ?_⟩
+  rw [Fin.strictMono_iff_lt_succ]
+  intro i
+  obtain h' | h' := (stdSimplex.monotone_apply s i.castSucc_le_succ).lt_or_eq
+  · exact h'
+  · simpa [Fin.ext_iff] using h h'
+
+lemma objEquiv_symm_mem_nonDegenerate_iff_mono {n d : ℕ} (f : ⦋d⦌ ⟶ ⦋n⦌) :
+    (objEquiv.{u} (m := (op ⦋d⦌))).symm f ∈ Δ[n].nonDegenerate d ↔ Mono f := by
```

Comment:
Removing `(m := (op ⦋d⦌))` here makes lean infers `m` as `op d`, breaking through the defeq `d = ⦋d⦌`, despite irreducibility of SimplexCategory. Do you understand why this happens?

Reply:
No, I do not understand what is happening here!